lib/models/addon.js

```diff
@@ -371,16 +387,36 @@ let addonProto = {
   },
 
   /**
-    Discovers all child addons of this addon and stores their names and
-    package.json contents in this.addonPackages as key-value pairs
-
-    @private
-    @method discoverAddons
+   * Discovers all child addons of this addon and an AddonInfo about
+   * each addon in this.addonPackages (keyed on addon name).
+   *
+   * Child addons include those from 'dependencies' (not devDependencies)
+   * and in-repo addons
+   *
+   * Any packageInfos that we find that are marked as not valid are excluded.
+   *
+   * @private
+   * @method discoverAddons
    */
   discoverAddons() {
-    let addonsList = this.addonDiscovery.discoverChildAddons(this);
+    let pkgInfo = this.packageInfoCache.getEntry(this.root);
+
+    if (pkgInfo) {
+      let self = this;
+      let addonPackageList = pkgInfo.discoverAddonAddons();
+      this.addonPackages =
+          pkgInfo.generateAddonPackages(addonPackageList,
+            addonInfo => (self.shouldIncludeChildAddon && !self.shouldIncludeChildAddon(addonInfo)));
```

Comment:
This appears to have introduced a regression, see originaly:

https://github.com/ember-cli/ember-cli/pull/7729/files#diff-6a115371fcdc66c67f9710569fa7c15eL75

Reply:
Actually, the double negation is occurring, this split over two different files. This may be fine.